pkg/kubectl/rolling_updater.go

```diff
@@ -208,13 +208,20 @@ func (r *RollingUpdater) Update(config *RollingUpdaterConfig) error {
 	if err != nil {
 		return err
 	}
-	// Further validation.
-	if maxUnavailable == 0 && maxSurge == 0 {
+	// Validate maximums.
+	if desired > 0 && maxUnavailable == 0 && maxSurge == 0 {
 		return fmt.Errorf("one of maxSurge or maxUnavailable must be specified")
 	}
 	// The minumum pods which must remain available througout the update
 	// calculated for internal convenience.
-	minAvailable := original - maxUnavailable
+	minAvailable := desired - maxUnavailable
```

Comment:
Sorry should have pointed it out earlier: We should set minAvailable to min(0, desired - maxUnavailable).